lib/auth/webauthnwin/ctypes_windows.go

```diff
@@ -0,0 +1,329 @@
+	dwTimeoutMilliseconds uint32
+	// Allowed Credentials List.
+	CredentialList webauthnCredentials
+	// Optional extensions to parse when performing the operation.
+	Extensions webauthnExtenstions
+	// Optional. Platform vs Cross-Platform Authenticators.
+	dwAuthenticatorAttachment uint32
+	// User Verification Requirement.
+	dwUserVerificationRequirement uint32
+	// Flags
+	dwFlags uint32
+
+	//
+	// The following fields have been added in WEBAUTHN_AUTHENTICATOR_GET_ASSERTION_OPTIONS_VERSION_2
+	//
+
+	// Optional identifier for the U2F AppId. Converted to UTF8 before being hashed. Not lower cased.
+	pwszU2fAppId *uint16
+	// If the following is non-NULL, then, set to TRUE if the above pwszU2fAppid was used instead of
+	// PCWSTR pwszRpId;
+	pbU2fAppId uint32
+
+	//
+	// The following fields have been added in WEBAUTHN_AUTHENTICATOR_GET_ASSERTION_OPTIONS_VERSION_3
+	//
+
+	// Cancellation Id - Optional - See WebAuthNGetCancellationId
+	pCancellationId *windows.GUID
+
+	//
+	// The following fields have been added in WEBAUTHN_AUTHENTICATOR_GET_ASSERTION_OPTIONS_VERSION_4
+	//
+
+	// Allow Credential List. If present, "CredentialList" will be ignored.
+	pAllowCredentialList *webauthnCredentialList
+
+	//
+	// The following fields have been added in WEBAUTHN_AUTHENTICATOR_GET_ASSERTION_OPTIONS_VERSION_5
+	//
+
+	dwCredLargeBlobOperation uint32
+	// Size of pbCredLargeBlob
+	cbCredLargeBlob uint32
+	pbCredLargeBlob *byte
+}
+
+type webauthnAssertion struct {
+	dwVersion uint32
+
+	// Size of cbAuthenticatorData.
+	cbAuthenticatorData uint32
+	// Authenticator data that was created for this assertion.
+	pbAuthenticatorData *byte
+
+	// Size of pbSignature.
+	cbSignature uint32
+	// Signature that was generated for this assertion.
+	pbSignature *byte
+
+	// Credential that was used for this assertion.
+	Credential webauthnCredential
+
+	// Size of User Id
+	cbUserId uint32
+	// UserId
+	pbUserId *byte
+
+	//
+	// Following fields have been added in WEBAUTHN_ASSERTION_VERSION_2
+	//
+
+	Extensions webauthnExtenstions
+
+	// Size of pbCredLargeBlob
+	cbCredLargeBlob       uint32
+	pbCredLargeBlob       *byte
+	dwCredLargeBlobStatus uint32
+}
+
+type webauthnX5C struct {
+	// Length of X.509 encoded certificate
+	cbData uint32
+	// X.509 DER encoded certificate bytes
+	pbData *byte
+}
+
+type webauthnCommonAttestation struct {
+	dwVersion uint32
+
+	// Hash and Padding Algorithm
+	//
+	// The following won't be set for "fido-u2f" which assumes "ES256".
+	pwszAlg *uint16
+	lAlg    int32 // COSE algorithm
+
+	// Signature that was generated for this attestation.
+	cbSignature uint32
+	pbSignature *byte
+
+	// Following is set for Full Basic Attestation. If not, set then, this is Self Attestation.
+	// Array of X.509 DER encoded certificates. The first certificate is the signer, leaf certificate.
+	cX5c uint32
+	pX5c *webauthnX5C
+
+	// Following are also set for TPM
+	pwszVer    *uint16 // L"2.0"
```

Comment:
nit: I don't think that C casting is needed here.
```suggestion
	pwszVer    *uint16 // "2.0"
```